modules/platform/nuxeo-platform-document-routing/nuxeo-routing-rest-api/src/main/java/org/nuxeo/ecm/restapi/server/jaxrs/routing/WorkflowObject.java

```diff
@@ -67,12 +67,17 @@ protected void initialize(Object... args) {
 
     @POST
     public Response createWorkflowInstance(WorkflowRequest workflowRequest) {
-        final String workflowInstanceId = documentRoutingService.createNewInstance(
-                workflowRequest.getWorkflowModelName(), workflowRequest.getAttachedDocumentIds(),
-                workflowRequest.getVariables(), ctx.getCoreSession(), true);
-        DocumentModel workflowInstance = getContext().getCoreSession().getDocument(new IdRef(workflowInstanceId));
-        DocumentRoute route = workflowInstance.getAdapter(DocumentRoute.class);
-        return Response.ok(route).status(Status.CREATED).build();
+        if (documentRoutingService.canCreateInstance(ctx.getCoreSession(), workflowRequest.getAttachedDocumentIds(),
+                workflowRequest.getWorkflowModelName())) {
+            final String workflowInstanceId = documentRoutingService.createNewInstance(
+                    workflowRequest.getWorkflowModelName(), workflowRequest.getAttachedDocumentIds(),
+                    workflowRequest.getVariables(), ctx.getCoreSession(), true);
+            DocumentModel workflowInstance = getContext().getCoreSession().getDocument(new IdRef(workflowInstanceId));
+            DocumentRoute route = workflowInstance.getAdapter(DocumentRoute.class);
+            return Response.ok(route).status(Status.CREATED).build();
+        } else {
+            return Response.status(Status.FORBIDDEN).build();
```

Comment:
I'm not sure about `FORBIDDEN` here, does @nelsonsilva or @nuxeo/platform have an opinion with respect to REST API best practices? Maybe a 400?

Reply:
@michaelva after some discussions it seems that `Status.BAD_REQUEST` would be better.